data_transfer/jobs/shared.py

```diff
@@ -26,3 +45,68 @@ def batch_upload_data() -> None:
                 record.is_uploaded = True
                 update_record(record)
             dmpy.rm_local_data(zip_path)
+
+
+def prepare_data_folders() -> None:
+    """
+    Checks folders present in 'data/upload/ are finished
+    and moves them into the upload folder in the format:
+
+        DEVICEID-PATIENTID-STARTWEAR-ENDWEAR
+    """
+
```

Comment:
I've tried to come up with a device-specific `batch_upload_data`, which is either:
- add a `regex` to determine which folders include the device acronym
- move folders to upload into device specific folders, e.g. `uploading/BTF/`

I find the first dodgy, the second acceptable. Do we want to do this in order to stage the upload better? Or leave as is (upload for all devices) and just make sure pipelines are executed sequentially?

Reply:
> I find the first dodgy, the second acceptable. Do we want to do this in order to stage the upload better? Or leave as is (upload for all devices) and just make sure pipelines are executed sequentially?

Yeah, might be better to have them separate as pipelines will likely run independently. Another option could be to filter folders before zipping based on device type since it's in the folder name and would be passed as parameter (in the none-sequential, device specific `batch_upload_data` method)?

```python
folders_to_upload = [p for p in config.upload_folder.iterdir() if p.is_dir() and NAME in p]
```

Reply:
Yes, sorry - that's what I meant with the `regex`. I just felt that it was slightly dodgy - as the acronym might just possibly be in the device_id or patient_id (although very very unlikely). But happy to include if you feel that is just never going to happen!

Reply:
Great point 👍🏼🚀  You're right, option two (`uploading/device_id`) would be ideal and likely simplifies the code in prepare too